### ISSUE TITLE
Issue with device - New Stake Outlets (with stake light) not turning on Outlet

### ISSUE THREAD

**Issue by tgskiman** [jdeath/Hubspace-Homeassistant#117]
I am having issues with the new outdoor stake outlet that includes the light on the stake itself. I am able to turn on and off the light on the stake, but am unable to turn on the outlets. When I do change the state, it just goes back to the previous state after a few seconds. The older stake outlets without lights continue to work as expected. I'm happy to provide the raw file if necessary. Thank you for the help and for all the work you have put into this!

[_dump_hs_devices.json](https://github.com/user-attachments/files/18013237/_dump_hs_devices.json)

**Comment by Expl0dingBanana** [comment]
Thanks for providing the dump (and not the raw dump - that may have been a mistake on my part)! Can you provide the logs where you enable the spike and it turns itself back off? Curious if HA is sending the off signal.

**Comment by tgskiman** [comment]
[Here is the debug log.](https://github.com/user-attachments/files/18015536/home-assistant_hubspace_2024-12-05T00-03-38.707Z.log) The outlet was on so I turned it off. I don't see anything setting it back to off but does look like it polls the queries the service after that and that is when it resets. I also included a screenshot of the logbook below. Let me know if you are looking for something else. Thanks again!!

![Screenshot 2024-12-04 190856](https://github.com/user-attachments/assets/b7f3b6fc-0473-4248-a249-8d7b705b5f38)

**Comment by Expl0dingBanana** [comment]
I see a lot of power stakes in the data dump. How many do you have connected? I see 7 "Smart Stake Timer" and 2 "Stake Outlets". I am wondering if we are toggling the timers rather than the outlets. Can you control individual outlets, or is it an off / off for the whole stake?

**Comment by tgskiman** [comment]
You are correct; I have 7. 5 are the "old style" ([Old product description](https://www.homedepot.com/p/Defiant-Smart-Outdoor-4-ft-6-Outlet-Power-Stake-Timer-Powered-by-Hubspace-LH-7-1-4FT/324832048)) with just the outlets that work fine and two are the "new ones" ([New product description](https://www.homedepot.com/p/Smart-Outdoor-4-ft-6-Outlet-Power-Stake-Timer-with-Holiday-Lights-Projector-Powered-by-Hubspace-LH-7-6-4FT/329191073)) that also include the light on the stake. It is on/off for the entire stake for both models). In the UI from Hubspace, the new stake has child controls for the light and the outlets as opposed to the single control with no children on the old style.

**Comment by tgskiman** [comment]
You are correct; I have 7. 5 are the "old style" ([Old product description](https://www.homedepot.com/p/Defiant-Smart-Outdoor-4-ft-6-Outlet-Power-Stake-Timer-Powered-by-Hubspace-LH-7-1-4FT/324832048)) with just the outlets that work fine and two are the "new ones" ([New product description](https://www.homedepot.com/p/Smart-Outdoor-4-ft-6-Outlet-Power-Stake-Timer-with-Holiday-Lights-Projector-Powered-by-Hubspace-LH-7-6-4FT/329191073)) that also include the light on the stake. It is on/off for the entire stake for both models). In the UI from Hubspace, the new stake has child controls for the light and the outlets as oppsed to the single control with no children on the old style.

**Comment by Expl0dingBanana** [comment]
Would it be possible to have it off, grab a data dump after a minute, then turn it on (via the app), wait a minute, and grab the dump?

**Comment by Expl0dingBanana** [comment]
Reviewing the JSON dump I see the following (with connecting some dots)

Old style:
```
        "model": "",
        "device_class": "power-outlet",
        "default_name": "Smart Stake Timer",
        "default_image": "smart-stake-timer-icon",
```


New Style:
``` 
        "model": "TBD",
        "device_class": "power-outlet",
        "default_name": "Stake Outlets",
        "default_image": "litesun-stake-timer-light-icon",
        
        "model": "TBD",
        "device_class": "power-outlet",
        "default_name": "Smart Stake Timer",
        "default_image": "litesun-stake-timer-light-icon",
```

Ill be curious to see if its timer or outlets that switches

**Comment by tgskiman** [comment]
I think you hit it on the head! Right now I see both elements in the screen shot above. I believe the "Smart Stake Timer" corelates to the working "Driveway West Stake Light" which does turn on the light on the stake. That would leave the actual outlets (called "driveway west outlets" in my screenshot) that are not working as the "Stake Outlets." For the dumps, do you want the raw or devices files? Thank you!!

**Comment by Expl0dingBanana** [comment]
Device files should be good! Raw includes things like GPS coords and we
don't want those

**Comment by tgskiman** [comment]
OK. I ended up making three files:

1. Baseline with the stake and stake light off: [All_Off.json](https://github.com/user-attachments/files/18104593/All_Off.json)
2. Stake outlet on (current scenario not working): [Outlet_On.json](https://github.com/user-attachments/files/18104601/Outlet_On.json)
3. Stake light on (currently working): [Light_On.json](https://github.com/user-attachments/files/18104602/Light_On.json)

For the outlet, I see the state change in line 2034 and for the light it is in 5698.

One other interesting thing I noted is that there is a "primary" node that is set on line 5515 in each case. I believe this is tied to (what I would call a "virtual master" switch) that is used in the app to show that either are on (or if you select when both are off, will turn both outlet and light on). Here is an image of the interface:
![image](https://github.com/user-attachments/assets/ef376402-8960-4b3d-987a-b508561a3cd6)

Thanks again!!

**Comment by Expl0dingBanana** [comment]
Reviewing device dumps with lots of devices and incorrect device links is rough 😅 Can you download the pre-release `dev-anon-data` and regenerate the data? Sorry to have you do this twice!

I am thinking once the `main` is identified, we ignore that element (similar to what I did with fans) and it may work better.

**Comment by tgskiman** [comment]
Here are the updated filesusing the dev-anon-data tagged pre-release: 

- [All_Offv2.json](https://github.com/user-attachments/files/18132554/All_Offv2.json)
- [Outlet_Onv2.json](https://github.com/user-attachments/files/18132555/Outlet_Onv2.json)
- [Light_Onv2.json](https://github.com/user-attachments/files/18132557/Light_Onv2.json)

Thank you!

**Comment by Expl0dingBanana** [comment]
I think I have a dirty fix for the power stake. I will need to revisit main device / supporting device and create actual logic. I was able to get away with it for ceiling fans because it was a standalone class 😅 

Can you try the pre-release `dev-power-stake-fix`?

**Comment by tgskiman** [comment]
Unfortanetly no joy. It behaved the same for the new stakes outlets and broke the old style, so they lost their switch entities. Thanks for the try/work though!

**Comment by Expl0dingBanana** [comment]
@tgskiman Can you try the new pre-release, dev-aiohubspace? I changed how parent / child relations are mapped and I think it may solve the issue.

**Comment by tgskiman** [comment]
Hey @Expl0dingBanana sorry for the delay in my response. I couldn't get the branch to work as expected, but I can happily say that with the latest release, everything is now working as expected!! Thank you!! I really appreciate all the work and personal time you are putting into this project. Side note, I just saw services are back; I really appreciate that one too!! Again, thank you!

**Comment by Expl0dingBanana** [comment]
Glad its working! Ill close this issue since the problem seems to be resolved.
